src/compiler/utils.jl

```diff
@@ -1,22 +1,30 @@
 
-function get_function!(builderF, mod, name)
```

Comment:
The goal of the design here was that we didn't need to create the FunctionType everytime.

Reply:
We do, however, regardless, need the FT to ensure the right type is returned if the current function has the wrong functiontype (eg returns i64 rather than i8*)